worker/uniter/upgradecharmprofile/resolver.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgradecharmprofile
+
+import (
+	"errors"
+
+	"github.com/juju/juju/core/lxdprofile"
+	"github.com/juju/juju/worker/uniter/operation"
+	"github.com/juju/juju/worker/uniter/remotestate"
+	"github.com/juju/juju/worker/uniter/resolver"
+	"github.com/juju/loggo"
+)
+
+var logger = loggo.GetLogger("juju.worker.uniter.upgradecharmprofile")
+
+// ErrDoNotProceed is used to distinguish behaviour from
+// resolver.ErrNoOperation. i.e do not run any operations versus
+// this resolver has no operations to run.
+var ErrDoNotProceed = errors.New("do not proceed")
```

Comment:
Is it worth pulling this out from here and _upgradeseries_, and making it common?

Reply:
I'll have a think about it, my first reaction is no, because I want to ensure uniqueness of errors, but as long as the errors can't impact each other. It does make sense to do so - as it seems to be a good practice.